File: src/_1ProgrammingBasicsWithJavaMay2023/_2ConditionalStatements/_3MoreExercise/_06FlowerShop.java
package _1ProgrammingBasicsWithJavaMay2023._2ConditionalStatements._3MoreExercise;

import java.util.Scanner;

public class _06FlowerShop {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int numberOfMagnolias = Integer.parseInt(scanner.nextLine());
        int numberOfHyacinths = Integer.parseInt(scanner.nextLine());
        int numberOfRoses = Integer.parseInt(scanner.nextLine());
        int numberOfCacti = Integer.parseInt(scanner.nextLine());
        double priceOfGift = Double.parseDouble(scanner.nextLine());
        double priceOfMagnolias = numberOfMagnolias * 3.25;
        double priceOfHyacinths = numberOfHyacinths * 4.00;
        double priceOfRoses = numberOfRoses * 3.50;
        double priceOfCacti = numberOfCacti * 8.00;
        double priceOfFlowers = priceOfMagnolias + priceOfHyacinths + priceOfRoses + priceOfCacti;
        double taxes = priceOfFlowers * 0.05;
        double netMoney = priceOfFlowers - taxes;
        double diff = Math.abs(netMoney-priceOfGift);
        if (netMoney < priceOfGift) {
            double moneyNeed = Math.ceil(diff);
            System.out.printf("She will have to borrow %.0f leva.", moneyNeed);
        } else {
            double moneyLeft = Math.floor(diff);
            System.out.printf("She is left with %.0f leva.", moneyLeft);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_2ConditionalStatements/_3MoreExercise/_06FlowerShop.js
function flowerShop(numberOfMagnolia, numberOfHyacinth, numberOfRose, numberOfCactus, priceForGift) {
    const PRICE_PER_MAGNOLIA = 3.25;
    const PRICE_PER_HYACINTH = 4.00;
    const PRICE_PER_ROSE = 3.50;
    const PRICE_PER_CACTUS = 8.00;

    const TAXES = 0.05;

    const priceForMagnolia = numberOfMagnolia * PRICE_PER_MAGNOLIA;
    const priceForHyacinth = numberOfHyacinth * PRICE_PER_HYACINTH;
    const priceForRose = numberOfRose * PRICE_PER_ROSE;
    const priceForCactus = numberOfCactus * PRICE_PER_CACTUS;

    const totalPrice = priceForMagnolia + priceForHyacinth + priceForRose + priceForCactus;
    const totalPriceAfterTax = totalPrice - (totalPrice * TAXES);

    const moneyLeft = Math.floor(totalPriceAfterTax - priceForGift);
    const moneyNeeded = Math.ceil(priceForGift - totalPriceAfterTax);

    if (totalPriceAfterTax >= priceForGift) {
        console.log(`She is left with ${moneyLeft} leva.`);
    } else {
        console.log(`She will have to borrow ${moneyNeeded} leva.`);
    }
}